src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -3154,39 +3190,87 @@ trait Types
             //
             // A more "natural" unifier might be M[t] = [t][t => t]. There's lots of scope for
             // experimenting with alternatives here.
-
             val tpSym = tp.typeSymbolDirect
-            val rightToLeft = tpSym.hasAnnotation(definitions.unifyRightToLeftClass)
+            val rightToLeft = tpSym.annotations.exists(_ matches definitions.unifyRightToLeftClass)
```

Comment:
What's going on here? It looks like you blown away some of the improvements that Jason made to my original version?